lib/api/controllers/funnelController.js

```diff
@@ -200,16 +201,19 @@ class FunnelController {
       return -1;
     }
 
-    let result = 0;
+    // if the connection is closed there is no need to execute the request
+    // => discarding it
+    if (!this.kuzzle.router.isConnectionAlive(request.context)) {
+      delete this.pendingRequests[request.id];
+      return 0;
+    }
+
     this.checkRights(request)
       .then(modifiedRequest => this.processRequest(modifiedRequest))
       .then(processResult => callback(null, processResult))
-      .catch(err => {
-        this._executeError(err, request, true, callback);
-        result = 1;
-      });
+      .catch(err => this._executeError(err, request, true, callback));
```

Comment:
I've removed these lines because `result` is set to `1` asynchronously, but its value is returned before it can be changed by the promise result function.
Ergo, in its current state, `execute` always returns `0`.

Moreover, at this point, the result status of `execute` doesn't matter: `-1` is only used when no free request slot has been found, so that the request can be queued to be executed at a later time.